compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Util.hs

```diff
@@ -189,15 +190,20 @@ _TOptional = prism' TOptional $ \case
   _ -> Nothing
 
 _TUpdate :: Prism' Type Type
-_TUpdate = prism' TList $ \case
+_TUpdate = prism' TUpdate $ \case
```

Comment:
I think this is an unrelated bug fix

Reply:
Can we get tests for these? I’m actually somewhat surprised we didn’t already hit this. It seems to be used in the DAML-LF typechecker and afaict it will error if the prism does not match so it shouldn’t be too hard to make a case that fails without this bugfix.

Reply:
IIRC, we only use them as getters. That's why the problem never showed up. Maybe we can just delete them?